src/identifier.py

```diff
@@ -0,0 +1,124 @@
+
+from .extractor import Extractor
+from .encoder import Encoder
+from deepface.commons import functions, distance as dst
+from .utils import euclidian_l2
+import os
+from PIL import Image
+import numpy as np
+import math
+from viam.logging import getLogger
+
+LOGGER = getLogger(__name__)
+
+class Identifier:
+    def __init__(self,
+                 detector_backend:str,
+                 extraction_threshold:float, 
+                 grayscale:bool, 
+                 enforce_detection:bool, 
+                 align:bool,
+                 model_name:str,
+                 normalization:str,
+                 dataset_path: str, 
+                 labels_directories:dict, 
+                 distance_metric_name:str, 
+                 identification_threshold:float):
+        
+        self.model_name = model_name
+        target_size = functions.find_target_size(model_name=model_name)
+        
+        self.extractor = Extractor(target_size=target_size, 
+                                   detector_backend=detector_backend,
+                                   extraction_threshold = extraction_threshold, 
+                                   grayscale=grayscale,
+                                   enforce_detection=enforce_detection, 
+                                   align=align)
+        
+        self.encoder = Encoder(model_name = model_name,
+                               align=align,
+                               normalization=normalization)
+        
+        self.dataset_path = dataset_path
+        self.labels_directories = labels_directories
+        self.model_name = model_name
+        self.known_embeddings = dict()
+        if distance_metric_name == "cosine":
+            self.distance = dst.findCosineDistance
+        if distance_metric_name == "euclidean":
+            self.distance = dst.findEuclideanDistance
+        elif distance_metric_name == "euclidean_l2":
+            self.distance = euclidian_l2
+        
+        if identification_threshold is None:
+            self.identification_threshold = dst.findThreshold(model_name, distance_metric_name)
+        else:
+            self.identification_threshold = identification_threshold
+        
+    def compute_known_embeddings(self):
+        for label in self.labels_directories:
+            label_path = self.dataset_path+"/"+self.labels_directories[label]
+            embeddings = []
+            for file in os.listdir(label_path):
+                if (
+                    (".jpg" in file.lower())
+                    or (".jpeg" in file.lower())
+                    or (".png" in file.lower())
+                ):
```

Comment:
emit a warning in an else statement saying that the file did not end in ".jpeg", ".jpg", or ".png"